File: Codes/Garnetwzy/114.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    void flatten(TreeNode* root) {
        help(root);
    }
    
    TreeNode* help(TreeNode* root) {
        if(root == NULL)
            return NULL;
        if(root && root->left == NULL && root->right == NULL)
            return root;
        TreeNode *cLeft = help(root->left);
        TreeNode *cRight = help(root->right);
        if(cLeft) {
            root->left = NULL;
            root->right = cLeft;
            while(cLeft->right) {
                cLeft = cLeft->right;
            }
            cLeft->right = cRight;
        }
        return root;
    }
};

File: codes/HellDream/114.java
    1 	class Solution {
    2 	    public void flatten(TreeNode root) {
    3 	        if(root==null) return;
    4 	        if(root.left!=null) flatten(root.left);
    5 	        if(root.right!=null) flatten(root.right);
    6 	        TreeNode right = root.right;
    7 	        root.right = root.left;
    8 	        root.left = null;
    9 	        while(root.right!=null) root = root.right;
   10 	        root.right = right;
   11 	    }
   12 	}
